### ISSUE TITLE
[Pods 2.8 Bug] Block Editor not used for New CPT

### ISSUE THREAD

**Issue by CosmikCarrot** [pods-framework/pods#5936]
## Pods 2.8 version

2.8.0-b-1

## Description

I'm not sure if this is a bug, or I haven't setup a CPT correctly.

After creating a new CPT, the block editor is not viewable on a new Page.
See - https://www.screencast.com/t/qFuby1Qbn

## Testing instructions

1. Go to add New CPT.
2. Click on - Add New
3. View Editor being used. Not Block Editor.

## Expected behavior

I'm was expecting to see the new Block Editor.

## Screenshots / screencast

See - https://www.screencast.com/t/14TKLPeYBEUb

## Other super helpful information

<details>
<summary>WordPress Environment</summary>

WP 5.6.1
  
```
`
### wp-core ###

version: 5.6.1
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: undefined
environment_type: production
user_count: 15
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /home/customer/www/cosmikcarrot.com/public_html
wordpress_size: 67.70 MB (70985452 bytes)
uploads_path: /home/customer/www/cosmikcarrot.com/public_html/wp-content/uploads
uploads_size: 189.99 MB (199215826 bytes)
themes_path: /home/customer/www/cosmikcarrot.com/public_html/wp-content/themes
themes_size: 6.71 MB (7036995 bytes)
plugins_path: /home/customer/www/cosmikcarrot.com/public_html/wp-content/plugins
plugins_size: 170.63 MB (178917298 bytes)
database_size: 53.59 MB (56197120 bytes)
total_size: 488.62 MB (512352691 bytes)

### wp-dropins (1) ###

object-cache.php: true

### wp-active-theme ###

name: Stackable (stackable)
version: 1.0.4
author: Cosmik Carrot
author_website: https://www.cosmikcarrot.com
parent_theme: none
theme_features: core-block-patterns, align-wide, automatic-feed-links, title-tag, post-thumbnails, custom-logo, menus, html5, custom-background, eventbrite, woocommerce, custom-header, infinite-scroll, featured-content, jetpack-responsive-videos, jetpack-social-menu, jetpack-testimonial, jetpack-content-options, editor-style, editor-color-palette, widgets
theme_path: /home/customer/www/cosmikcarrot.com/public_html/wp-content/themes/stackable
auto_update: Disabled

### wp-themes-inactive (2) ###

Twenty Twenty: version: 1.6, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 1.1, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (29) ###

BackupBuddy: version: 8.7.2.0, author: iThemes, Auto-updates disabled
Cloudflare: version: 4.1.0, author: Cloudflare, Inc., Auto-updates disabled
Email Templates: version: 1.3.2.1, author: Damian Logghe, Auto-updates disabled
GAinWP Google Analytics Integration for WordPress: version: 5.4.6, author: IntelligenceWP, Auto-updates enabled
GitHub Updater: version: 9.9.8, author: Andy Fragen, Auto-updates disabled
GP Post Content Merge Tags: version: 1.2.3, author: Gravity Wiz, Auto-updates disabled
Gravity Forms: version: 2.4.22, author: Gravity Forms, Auto-updates enabled
Gravity Forms + Stripe: version: 2.0.0.0.beta5, author: gravity+, Auto-updates disabled
Gravity Forms + Stripe (More): version: 2.0.0.0.beta17, author: gravity+, Auto-updates disabled
Gravity Forms Currency Selector: version: 1.3.1, author: gravity+, Auto-updates disabled
Gravity Forms Signature Add-On: version: 4.0, author: Gravity Forms, Auto-updates disabled
Gravity Forms Zapier Add-on: version: 3.3, author: Gravity Forms, Auto-updates disabled
Gravity PDF: version: 5.3.3, author: Gravity PDF, Auto-updates disabled
Gravity Perks: version: 2.2.2, author: Gravity Wiz, Auto-updates disabled
If Menu - Visibility control for menu items: version: 0.16.2, author: Layered, Auto-updates disabled
iThemes Security Pro: version: 6.8.3, author: iThemes, Auto-updates disabled
iThemes Sync: version: 2.1.8, author: iThemes, Auto-updates disabled
Login Logout Menu: version: 1.1.0, author: WPBrigade, Auto-updates disabled
MemberPress Business Edition: version: 1.9.10, author: Caseproof, LLC, Auto-updates disabled
MemberPress Toolbox - Account Navigation Tabs: version: 1.0.6, author: MemberPress Toolbox, Auto-updates disabled
Pods - Custom Content Types and Fields: version: 2.8.0-b-1, author: Pods Framework Team, Auto-updates disabled
Safe SVG: version: 1.9.9, author: Daryll Doyle, Auto-updates disabled
SendGrid: version: 1.11.8, author: SendGrid, Auto-updates disabled
SG Optimizer: version: 5.7.14, author: SiteGround, Auto-updates disabled
Stackable - Gutenberg Blocks (Premium): version: 2.13.4, author: Gambit Technologies, Inc (latest version: 2.14.0), Auto-updates disabled
User Switching: version: 1.5.6, author: John Blackbourn & contributors, Auto-updates disabled
WPCosmik Carrot Custom Functions Plugin: version: 1.0, author: Cosmik Carrot, Auto-updates disabled
WP Review Slider Pro (Premium): version: 11.0.9.5, author: LJ Apps, Auto-updates disabled
Yoast SEO Premium: version: 15.8, author: Team Yoast, Auto-updates disabled

### wp-plugins-inactive (6) ###

SEMrush SEO Writing Assistant: version: 1.1.1, author: SEMrush, Auto-updates disabled
Site Kit by Google: version: 1.25.0, author: Google, Auto-updates disabled
snip - The Rich Snippets & Structured Data Plugin: version: 2.19.8, author: floriansimeth, Auto-updates disabled
WP All Import Pro: version: 4.6.4, author: Soflyy, Auto-updates disabled
WP Landing Kit: version: 1.1.2, author: <a href="https://circa75.co/">Circa75</a> & <a href="https://hookturn.io">Hookturn</a>, Auto-updates disabled
WP Rollback: version: 1.7.1, author: Impress.org, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
file_uploads: File uploads is turned off
post_max_size: 256M
upload_max_filesize: 256M
max_effective_size: 256 MB
max_file_uploads: 20
gd_version: 2.2.5
ghostscript_version: 9.27

### wp-server ###

server_architecture: Linux 3.12.18-clouder0 x86_64
httpd_software: Apache
php_version: 7.3.27 64bit
php_sapi: apache2handler
max_input_variables: 3000
time_limit: 120
memory_limit: 768M
max_input_time: 120
upload_max_filesize: 256M
php_post_max_size: 256M
curl_version: 7.66.0 OpenSSL/1.1.1g-fips
suhosin: false
imagick_availability: false
pretty_permalinks: true
htaccess_extra_rules: true

### wp-database ###

extension: mysqli
server_version: 5.7.32-35-log
client_version: mysqlnd 5.0.12-dev - 20150407 - $Id: 7cc7cc96e675f6d72e5cf0f267f48e167c2abb23 $

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /home/customer/www/cosmikcarrot.com/public_html/wp-content
WP_PLUGIN_DIR: /home/customer/www/cosmikcarrot.com/public_html/wp-content/plugins
WP_MAX_MEMORY_LIMIT: 768M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_LOCAL_DEV: undefined
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
mu-plugins: writable

### pods ###

pods-server-software: Apache
pods-user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36 OPR/73.0.3856.344
pods-session-save-path: /tmp
pods-session-save-path-exists: Yes
pods-session-save-path-writable: Yes
pods-session-max-lifetime: 1440
pods-opcode-cache-apc: No
pods-opcode-cache-memcached: No
pods-opcode-cache-opcache: Yes
pods-opcode-cache-redis: No
pods-object-cache-apc: No
pods-object-cache-apcu: No
pods-object-cache-memcache: No
pods-object-cache-memcached: Yes
pods-object-cache-redis: No
pods-memory-current-usage: 19.514M
pods-memory-current-usage-real: 6.000M
pods-network-wide: No
pods-install-location: /home/customer/www/cosmikcarrot.com/public_html/wp-content/plugins/pods-release-2.8/
pods-developer: No
pods-tableless-mode: No
pods-relationship-table-enabled: Yes
pods-light-mode: No
pods-strict: No
pods-allow-deprecated: Yes
pods-api-cache: Yes
pods-shortcode-allow-evaluate-tags: No

`
```

</details>

<details>
<summary>Pods Package Export</summary>

```json
Copy and paste your Pods Admin > Migrate: Packages > Export and paste the code over this line.
```

</details>

**Comment by sc0ttkclark** [comment]
To confirm, have you turned on the REST API endpoint option for this pod?

We've thought about enabling REST API by default for new post types in the past.

**Comment by CosmikCarrot** [comment]
Doh, no the REST API endpoint option for the pod wasn't turned on.

Personally, it's a no brainer to enable the REST API by default, however I totally understand that there is a long list of people yet to move over to the block editor, plus plenty page builders out there. 
Maybe if 'disable gutenberg' plugin is detected, deactivate REST API. Just a suggestion.

Thanks @sc0ttkclark! This is going to be awesome now with the new block editor. Thanks for everyones hard work!

**Comment by CosmikCarrot** [comment]
@sc0ttkclark How do I get to display these page attributes that are currently missing for a new CPT?

See - https://www.screencast.com/t/HGIEC3ZP2y

Page template, Featured image, Discussion

**Comment by CosmikCarrot** [comment]
I've now managed to add the page template for the CPT -  https://docs.pods.io/faqs/how-to-page-template-from-theme-with-custom-post-types/

Featured Image support 
`/* Featured Image Support */
	add_post_type_support( 'pod_custom-post-type', 'thumbnail' );`

**Comment by sc0ttkclark** [comment]
There are a few CPT/CT options not showing right now. The new editor work does not yet support the checkbox group display in a dynamic way and that's something we will have to add support for before beta is done.

**Comment by sc0ttkclark** [comment]
I've fixed the 'Supports' stuff in `release/2.8` so those are now available to use for customizing the configs as expected. See Pods Admin > Edit Pods > Edit your pod > Advanced tab > Supports section.

**Comment by unknownnf** [comment]
This seems to be resolved now. Supports section is showing.
